mlir/lib/Conversion/VectorToLLVM/ConvertVectorToLLVM.cpp

```diff
@@ -278,9 +313,17 @@ class VectorGatherOpConversion
 
     // Resolve alignment.
     unsigned align;
-    if (failed(getMemRefAlignment(*getTypeConverter(), memRefType, align))) {
-      return rewriter.notifyMatchFailure(gather,
-                                         "could not resolve memref alignment");
+    if (useVectorAlignment) {
+      if (failed(getVectorAlignment(*this->getTypeConverter(), vType, align))) {
+        return rewriter.notifyMatchFailure(
+            gather, "could not resolve vector alignment");
+      }
+    } else {
+      if (failed(getMemRefAlignment(*this->getTypeConverter(), memRefType,
+                                    align))) {
+        return rewriter.notifyMatchFailure(
+            gather, "could not resolve memref alignment");
+      }
```

Comment:
Similar code is repeated twice - could this be moved to a dedicated helper? I would prioritise making the API self-documenting :) (e.g. `getVectorToLLVMAlignment(Operation *op, useVectorAlignment, align)` or something similar).

Reply:
I added a helper